tests/unit/test_routes/test_resource_retreival.py

```diff
@@ -0,0 +1,186 @@
+
+    # Default page size should be specified in PaginatorConfig
+    assert (len(resources['data']) == PaginatorConfig.per_page)
+
+    for resource in resources['data']:
+        assert (isinstance(resource.get('name'), str))
+        assert (resource.get('name'))
+        assert (isinstance(resource.get('url'), str))
+        assert (resource.get('url'))
+        assert (isinstance(resource.get('category'), str))
+        assert (resource.get('category'))
+        assert (isinstance(resource.get('languages'), list))
+    assert (resources['number_of_pages'] is not None)
+
+    ua = datetime.now() + timedelta(days=-7)
+    uaString = ua.strftime('%m-%d-%Y')
+    response = client.get(f"/api/v1/resources?updated_after={uaString}")
+    assert (response.status_code == 200)
+
+    # Test trying to get a page of results that doesn't exist
+    too_far = 99999999
+    response = client.get(
+        f"api/v1/resources?page_size=100&page={too_far}", follow_redirects=True)
+    assert (response.status_code == 404)
+    assert (
+        isinstance(response.json.get('errors').get(
+            get_error_code_from_status(response.status_code)), dict))
+    assert (
+        isinstance(response.json.get('errors').get(
+            get_error_code_from_status(response.status_code)).get('message'), str))
+
+
+def test_get_resources_post_date_failure(module_client):
+    client = module_client
+    # If updated_after date is after today, then should return a 422
+    ua = datetime.now() + timedelta(days=1)
+    uaString = ua.strftime('%m-%d-%Y')
+    response = client.get(f"/api/v1/resources?updated_after={uaString}")
+    assert (response.status_code == 422)
+    assert (isinstance(response.json.get('errors').get(
+        get_error_code_from_status(response.status_code)), dict))
+    assert (isinstance(response.json.get('errors').get(
+        get_error_code_from_status(response.status_code)).get('message'), str))
+
+
+def test_get_single_resource(module_client, module_db):
+    client = module_client
+
+    response = client.get('api/v1/resources/5')
+
+    # Status should be OK
+    assert (response.status_code == 200)
+
+    resource = response.json['data']
+    assert (isinstance(resource.get('name'), str))
+    assert (resource.get('name'))
+    assert (isinstance(resource.get('url'), str))
+    assert (resource.get('url'))
+    assert (isinstance(resource.get('category'), str))
+    assert (resource.get('category'))
+    assert (isinstance(resource.get('languages'), list))
+
+    assert (resource.get('id') == 5)
+
+
+def test_single_resource_out_of_bounds(module_client, module_db):
+    client = module_client
+
+    too_low = 0
+    too_high = 9999
+    response = client.get(f"api/v1/resources/{too_low}", follow_redirects=True)
+
+    assert (response.status_code == 404)
+
+    response = client.get(f"api/v1/resources/{too_high}", follow_redirects=True)
+
+    assert (response.status_code == 404)
+
+
+def test_get_favicon(module_client):
```

Comment:
This test and the 2 below it don't actually have to do with getting resources. If we need to, we can break this out into its own file as well. Maybe a "junk drawer" type of general tests that are sort of different from the actual routes.. Maybe even throw it in `test_utils.py` which seems to be kind of similar to what I'm describing